### ISSUE TITLE
Add an alias for `source-paths` configuration key

### ISSUE THREAD

**Issue by clrcrl** [dbt-labs/dbt#1607]
This is a pretty confusing key now that we have `sources`!
```
source-paths: ["models"]
```
Can we add an alias, `model-paths` for this key? And then I'll update the starter project accordingly.

While we are here, would also love if we could give the rest of these more sensible aliases, e.g.:
* data-paths --> seed-paths
* test-paths --> something that doesn't make me think this is where I define my `unique` and `not_null` tests (let me think on the best alias for this

**Comment by leahwicz** [comment]
@jtcohen6 lets move all these requests to #2659 and close this issue out. It will be easier to get these all done at the same time

**Comment by jtcohen6** [comment]
> Let's schedule the work together though

Agreed! I wouldn't mind consolidating them into one issue, whatever feels most straightforward for implementation.

**Comment by emmyoop** [comment]
@jtcohen6 I have a few questions on this

First, I updated `source-paths` to be `model-paths` which makes so much sense I love it.  I think we also need to add a clear exception on what's going on.  Current code catches it but it isn't super clear.  The error you get if you are using `source-paths` is:

```Project loading failed for the following reason:
Runtime Error
  at path []: Additional properties are not allowed ('source-paths' was unexpected)
```

I was thinking something like this (but perfectly happy with you rewording it) which makes it more clear something changed and users need to take action.

```
Project loading failed for the following reason:
Runtime Error
  The `source-paths` alias has been deprecated in favor of `models-paths`. Please update `dbt_project.yml` to reflect this change.
```

Next - should I do the same with data-paths and test-paths as suggested in the original task?
